src/PoolManager.sol

```diff
@@ -144,6 +146,13 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
         Lock.lock();
     }
 
+    /// @inheritdoc IPoolManager
+    function sync(Currency currency) public returns (uint256 balance) {
+        balance = currency.balanceOfSelf();
+        if (balance == 0) balance = ZERO_BALANCE;
```

Comment:
i'm wondering if this should go inside the Reserves file?
Then in PoolManager we just have like `setReserves` and `getReserves` and `Reserves.sol` defines `ZERO_BALANCE` and `ReservesMustBeSynced ` and handles the `if` statements internally? what do you think?

Reply:
hm i do then see than `getReserves` wouldnt be a good view function if it sometimes reverts...
just feels like `ZERO_BALANCE` should live in there not here

Reply:
mm the more i think about it the more im not sure 😆

Reply:
i like the idea of consolidating some logic to the Reserves library, especially for storing the ZERO_BALANCE constant

---

as for making `getReserves` revert or not, it might sense to do so. I'm more concerned of devs/builders/integrators assuming that `0` is a valid return

Reply:
hm I slightly lean that it should live in sync only because get/set I see are just the interfaces for writing to transient storage and don't need to/(probably shouldn't) handle logic that is specific to implementation details of the sync function.